<?xml version="1.0"?>

<launch>
	<arg name="robotName" value="bebop"/>
	<arg name="controller" value="di_it2_flc"/> <!-- pid, t1flc, si_it2_flc, di_it2_flc, fnn -->
	<arg name="noise" default="0.00"/>

	<remap from="/uav/command" to="/$(arg robotName)/command"/>
	<remap from="/uav/command_velocity" to="/$(arg robotName)/command_velocity"/>
	<remap from="/uav/position" to="/$(arg robotName)/position"/>
	<remap from="/uav/velocity" to="/$(arg robotName)/linear_velocity"/>
	<remap from="/uav/noise/position" to="/$(arg robotName)/noise/position"/>
	<remap from="/uav/odometry" to="/$(arg robotName)/odometry"/>
	<remap from="/uav/true_odometry" to="/$(arg robotName)/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/$(arg robotName)/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/$(arg robotName)/trajectory_velocity"/>
	<remap from="/uav/noise" to="/$(arg robotName)/noise"/>
	<remap from="/aruco_poses" to="/aruco/odometry"/>

	<node name="teleop_key" pkg="controllers" type="teleop_key" output="screen" args=""/>
	<node name="safe_bebop" pkg="controllers" type="safe_bebop" output="screen" args=""/>
	<node name="trajectory" pkg="controllers" type="trajectory" output="screen" args=""/>
	<!-- node name="noise" pkg="controllers" type="noise" output="screen" args="$(arg noise)"/ -->
	<node name="logger" pkg="controllers" type="logger" output="screen" args="2016.10.25-$(arg controller)-wind_0-1"/>

	<include file="$(find mocap_optitrack)/launch/mocap.launch"/>
	<include file="$(find controllers)/launch/$(arg controller).launch"/>
	<include file="$(find bebop_driver)/launch/bebop_node.launch"/>
	<!-- include file="$(find aruco_mapping)/launch/aruco_mapping.launch"/ -->

	<node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/>

	<!-- node name="rqt_image_view" pkg="rqt_image_view" type="rqt_image_view" args="/bebop/image_raw"/ -->
</launch>
